src/app/shared/directives/hide-if-not-authenticated.directive.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, Injectable, TemplateRef, ViewContainerRef } from '@angular/core';
+import { HideIfNotAuthenticatedDirective } from './hide-if-not-authenticated.directive';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { AuthenticationService } from '../services/authentication.service';
+
+describe('HideIfNotAuthenticatedDirective', () => {
+
+  @Injectable()
+  class MockService extends AuthenticationService {
+
+  }
+
+  @Component({
+    template: ''
+  })
+  class MockComponent {
+    constructor(public viewContainerRef: ViewContainerRef, public mockAuthService: MockService) {}
+    myTemplateRef!: TemplateRef<unknown>;
+  }
+
+  const elementRefMock = jasmine.createSpyObj('ElementRef', ['nativeElement']);
+  const rendererRefMock = jasmine.createSpyObj('Renderer2', ['nativeElement']);
+
+  let fixture: ComponentFixture<MockComponent>;
+
+  beforeEach( () => {
+    fixture = TestBed.configureTestingModule({
+      declarations: [MockComponent, HideIfNotAuthenticatedDirective],
+      providers: [
+        {provide: MockService}
+      ]
+    }).createComponent(MockComponent);
+  });
+
+  it('should create an instance', () => {
+    const mockTemplateRef = fixture.componentInstance.myTemplateRef;
```

Comment:
I recommend avoid using `fixture.componentInstance` to access the dependencies
You can create variables in the describe 
`let mockTemplateRef: TemplateRef<any>;
  let mockViewContainerRef: ViewContainerRef;
  let mockAuthService: AuthService;`
  
  Then inject them in the providers array
  `providers: [
        { provide: TemplateRef, useValue: {} },
        { provide: ViewContainerRef, useValue: {} },
        { provide: AuthService, useValue: {} }
      ]`
      
      And then inject the mock services
      `mockTemplateRef = TestBed.inject(TemplateRef);
    mockViewContainerRef = TestBed.inject(ViewContainerRef);
    mockAuthService = TestBed.inject(AuthService);

    directive = new HideIfNotAuthenticatedDirective(
      mockTemplateRef,
      mockViewContainerRef,
      mockAuthService
    );`

Reply:
thanks, need to examine the code, testing it's just the whole new universe, so difficult to embrace